### PR TITLE
Add boto3 recipe

### PR DESCRIPTION
**Pull request by proinsias**
- [x] Add `s3transfer` to `conda-forge`

### CODE REVIEW

File: recipes/boto3/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  script: python setup.py install --single-version-externally-managed --record record.txt
+  skip: True  # [win and py3k]
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - futures >=2.2.0,<4.0.0  # [py2k]
+    - botocore >=1.4.1,<1.5.0
+    - jmespath >=0.7.1,<1.0.0
```
Comment (jakirkham) [inline recipes/boto3/meta.yaml]: Do we need all of these during the build?

File: recipes/boto3/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{% set name = "boto3" %}
+{% set version = "1.3.1" %}
```
Comment (jakirkham) [inline recipes/boto3/meta.yaml]: Should we bump to 1.4.0 as that is [out]( https://pypi.python.org/pypi/boto3/1.4.0 )?

Comment (jakirkham) [inline recipes/boto3/meta.yaml]: The dependencies change slightly in that release too.

File: recipes/boto3/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+  skip: True  # [win and py3k]
```
Comment (jakirkham) [inline recipes/boto3/meta.yaml]: Please try building on Windows.

**Comment by proinsias** [comment]
The CircleCI build says:

`conda.resolve.NoPackagesFound: Packages/dependencies missing in current linux-64 channels: 
  - boto3 1.4.0 py27_0 -> s3transfer >=0.1.0,<0.2.0
  - s3transfer >=0.1.0,<0.2.0`

Why wouldn't the `s3transfer` build be compatible with python 2.7?

**Comment by proinsias** [comment]
@jakirkham – should be RTM.

File: recipes/boto3/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  license_family: MIT
+  license_file: LICENSE
+  summary: Amazon Web Services SDK for Python
+  description: Boto3 makes it easy to integrate you Python application, library or script
+    with AWS services. It allows Python developers to write softare that makes
+    use of services like Amazon S3 and Amazon EC2.
+
+extra:
+  recipe-maintainers:
+    - proinsias
```
Comment (jakirkham) [inline recipes/boto3/meta.yaml]: Any interest in this, @pmlandwehr ?

File: recipes/boto3/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  home: http://aws.amazon.com/sdk-for-python/
+  dev_url: https://github.com/boto/boto3
+  doc_url: https://boto3.readthedocs.org
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: Amazon Web Services SDK for Python
+  description: Boto3 makes it easy to integrate you Python application, library or script
+    with AWS services. It allows Python developers to write softare that makes
+    use of services like Amazon S3 and Amazon EC2.
```
Comment (jakirkham) [inline recipes/boto3/meta.yaml]: Interesting. I kind of like this metadata ordering. 😄

**Comment by jakirkham** [comment]
Will give Peter some time to respond. Otherwise am happy to merge. 👍

File: recipes/boto3/meta.yaml (hunk 4 above)
Comment (jakirkham) [inline recipes/boto3/meta.yaml]: Haven't heard back. So am going to go ahead and merge. Feel free to add yourself as a maintainer in the feedstock if you are interested.

**Comment by jakirkham** [comment]
Thanks @proinsias .

**Pull request merged**